pyspecProcScripts/correlation_alignment.py

```diff
@@ -52,8 +61,20 @@ def correl_align(s, align_phases=False,tol=1e-4,indirect_dim='indirect',fig_titl
                 the width of the Gaussian function used to frequency filter
                 the data in the calculation of the correlation function.
     """
-    ph1_len = len(s.getaxis('ph1'))
-    ph2_len = len(s.getaxis('ph2'))
+    logging.info(strm("Applying the correlation routine"))
+    if nScans:
+        phcycdims = [j for j in s.dimlabels if j.startswith('ph')]
+        indirect = set(s.dimlabels)-set(phcycdims)-set([direct])
+        indirect = [j for j in s.dimlabels if j in indirect]
+        nScans_len = len(s.getaxis('nScans'))
+        s.smoosh(indirect)
+    s.ift(list(signal_pathway.keys()))
+    ph_lens = []
+    signal_keys = list(signal_pathway)
+    signal_values = list(signal_pathway.values())
+    for x in range(len(signal_keys)):
+        ph_len = len(s.getaxis(signal_keys[x]))
+        ph_lens.append(ph_len)
```

Comment:
these could be rewritten as `ph_lens = [len(j) for j in s.getaxis(signal_keys[x])]`

I would eliminate the previous two lines and just do this
`ph_len = {j:ndshape(s)[j] for j in signal_pathway.keys()}`

You will want a dictionary below, otherwise, there is an undetermined order issue here.

Reply:
I can make the dictionary but the first guy is not working - I get this error 'TypeError: object of type 'numpy.float64' has no len()' which I can work on trouble shooting but while I adress the other issues I'm going to keep what works but keep this guy in there just commented out.

Reply:
So I looked through the code and ph_len and ph_lens is not used anywhere in the code and so I deleted this little loop. It still runs on my end without it.